saos-persistence/src/main/java/pl/edu/icm/saos/persistence/enrichment/model/JudgmentEnrichmentHash.java

```diff
@@ -0,0 +1,118 @@
+package pl.edu.icm.saos.persistence.enrichment.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.apache.commons.lang3.StringUtils;
+
+import pl.edu.icm.saos.persistence.common.DataObject;
+
+/**
+ * Entity containing hashes generated from enrichment tags related to judgments 
+ * (generated for it, see: {@link EnrichmentTag#getJudgmentId()} or 
+ * referring to it in their value, see: {@link EnrichmentTag#getValue()}).<br/>
+ * Hashes are used to determine if enrichment tags have changed 
+ * since the last processing of them.
+ * 
+ * @author madryk
+ */
+@Entity
+@Table(uniqueConstraints={@UniqueConstraint(name="tag_judgment_id_unique", columnNames={"judgmentId"})})
+@SequenceGenerator(name = "seq_judgment_enrichment_hash", allocationSize = 1, sequenceName = "seq_judgment_enrichment_hash")
+public class JudgmentEnrichmentHash extends DataObject {
+
+    private long judgmentId;
+    
+    private String oldHash;
+    
+    private String hash;
+    
+    private boolean processed;
+
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Updates hash value (see {@link #getHash()}). Marks judgment enrichment tags as not processed if
+     * hash value changed since last enrichment tags processing.
+     */
+    public void updateHash(String newHash) {
+        if (processed) {
+            this.oldHash = this.hash;
+            this.hash = newHash;
+            if (!StringUtils.equals(this.oldHash, this.hash)) {
+                processed = false;
+            }
+        } else {
+            this.hash = newHash;
```

Comment:
I wonder if the processed flag shouldn't be set here to true if this.hash == this.oldHash.